### PR TITLE
Added OpenVINO 2024.1.0 migration file

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
CC @xhochy @hmaarrfk

File: recipe/migrations/openvino202410.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+migrator_ts: 1712592149
+__migrator:
+  kind: version
+  migration_number: 1
+  bump_number: 1
+openvino:
+  - 2024.1.0
```
Comment (hmaarrfk) [inline recipe/migrations/openvino202410.yaml:7]:
```suggestion
  - 2024.1.0
libopenvino_dev:
  - 2024.1.0
```

sadly, you probably want to add a line for each and every one of your outputs just to make sure. i never asked why you had so many, but if it is conceivable that somebody depends on one of them at build time, then they should be included here.

We don't have a good way to deal with these "linked" dependencies to my knowledge.

See for example:
https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/5664/files

Comment (ilya-lavrenov) [reply recipe/migrations/openvino202410.yaml:7]: At build time we need only `libopenvino_dev` which introduces dependency on `libopenvino`, other outputs are plugins.

File: recipe/migrations/openvino202410.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
```
Comment (hmaarrfk) [inline recipe/migrations/openvino202410.yaml:7]:
```suggestion
```

if that is the case, we can delete this too. Unless yo uthink that some have erroneously depended on openvino at build time.

**Pull request merged**